test/unit/parse_tree_visitor_test.rb

```diff
@@ -159,6 +159,13 @@ def test_for_body
     )
   end
 
+  def test_for_range
+    assert_equal(
+      ["test"],
```

Comment:
Shouldn't it return start_obj too?

```suggestion
      [1, "test"],
```

Reply:
No, because visit only looks for VariableLookups.

https://github.com/Shopify/liquid/pull/1470/files#diff-22cc20d81292886f308182d7ab35637947cb1ed3f4ddae195e3c5931c0baeaf3L251-L252

Reply:
https://github.com/Shopify/liquid/blob/e86fe272599596e14d40a19ceb5ab7f5e7207017/test/unit/parse_tree_visitor_test.rb#L252-L260